File: Program's_Contributed_By_Contributors/Python_Programs/knapsack.py
w=[int(x) for x in input("enter weights: ").split()]
v=[int(x) for x in input("enter values: ").split()]
bag=int(input("enter bag capacity: "))
n=len(v)
def knapSack(bag,w,v,n):
    if n==0 or bag==0:
        return 0
    if (w[n-1] > bag):
        return knapSack(bag, w, v, n-1)
    else:
        return max(v[n-1] + knapSack(bag-w[n-1], w, v, n-1),knapSack(bag, w, v, n-1))
knapSack(bag,w,v,n)
# knapsack problem by ayushtripathi12103634

File: Program's_Contributed_By_Contributors/C++/knapsack.cpp
//Username- kajolkhairnar
//Description- Knapsack problem is the most important problem in list of greedy approch problems.
#include <iostream>
using namespace std;
int max(int x, int y) {
   return (x > y) ? x : y;
}
int knapSack(int W, int w[], int v[], int n) {
   int i, wt;
   int K[n + 1][W + 1];
   for (i = 0; i <= n; i++) {
      for (wt = 0; wt <= W; wt++) {
         if (i == 0 || wt == 0)
         K[i][wt] = 0;
         else if (w[i - 1] <= wt)
            K[i][wt] = max(v[i - 1] + K[i - 1][wt - w[i - 1]], K[i - 1][wt]);
         else
        K[i][wt] = K[i - 1][wt];
      }
   }
   return K[n][W];
}
int main() {
   cout << "Enter the number of items in a Knapsack:";
   int n, W;
   cin >> n;
   int v[n], w[n];
   for (int i = 0; i < n; i++) {
      cout << "Enter value and weight for item " << i << ":";
      cin >> v[i];
      cin >> w[i];
   }
   cout << "Enter the capacity of knapsack";
   cin >> W;
   cout << knapSack(W, w, v, n);
   return 0;
}
